src/lib.rs

```diff
@@ -126,15 +126,49 @@ pub trait Renderable: Send + Sync {
     fn render(&self, context: &mut Context) -> Result<Option<String>>;
 }
 
+pub type TemplateLocation = Path;
```

Comment:
If the intention is to abstract away where the snippets are coming from, I feel like keeping it as a `Str` rather than `Path` is ideal.

Also `Location` in the name doesn't quite sit right with me.  What do you think of something like `Name` or `Path`